internal/controller/environmentrequest_controller.go

```diff
@@ -116,19 +116,25 @@ func (r *EnvironmentRequestReconciler) reconcile(ctx context.Context, environmen
 	logger.V(1).Info("environment provider count", "active", len(environmentProviders.activeJobs), "successful", len(environmentProviders.successfulJobs), "failed", len(environmentProviders.failedJobs))
 
 	// Check providers availability
-	// TODO: Update status
 	providers := etosv1alpha1.Providers{
 		IUT:            environmentrequest.Spec.Providers.IUT.ID,
 		ExecutionSpace: environmentrequest.Spec.Providers.ExecutionSpace.ID,
 		LogArea:        environmentrequest.Spec.Providers.LogArea.ID,
 	}
 	if err := checkProviders(ctx, r, environmentrequest.Namespace, providers); err != nil {
+		meta.SetStatusCondition(&environmentrequest.Status.Conditions, metav1.Condition{Status: metav1.ConditionTrue, Type: StatusFailed, Message: "Provider check failed", Reason: err.Error()})
+		if _err := r.Status().Update(ctx, environmentrequest); _err != nil {
```

Comment:
It's better to log the original error and return `r.Status().Update(...)` here. Since any error returned will look like an error in the controller which this case is not. This is just a reconciliation error.

```
meta.SetStatusCondition(....)
logger.Error(err, "Error getting provider")
return r.Status().Update(...)
```